Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -979,13 +979,30 @@ open class ChatMessageListVC: _ViewController,
     }
 
     open func didTapActionOnAttachment(_ attachment: ChatMessageFileAttachment, at indexPath: IndexPath?) {
-        switch attachment.uploadingState?.state {
-        case .uploadingFailed:
-            client
-                .messageController(cid: attachment.id.cid, messageId: attachment.id.messageId)
-                .restartFailedAttachmentUploading(with: attachment.id)
-        default:
-            break
+        if let uploadingState = attachment.uploadingState {
+            switch uploadingState.state {
+            case .uploadingFailed:
+                client
+                    .messageController(cid: attachment.id.cid, messageId: attachment.id.messageId)
+                    .restartFailedAttachmentUploading(with: attachment.id)
+            default:
+                break
+            }
+        } else if let downloadingState = attachment.downloadingState, downloadingState.state == .downloaded {
+            guard let indexPath, let cell = listView.cellForRow(at: indexPath) else { return }
+            let activityViewController = UIActivityViewController(activityItems: [downloadingState.localFileURL], applicationActivities: nil)
+            // TODO: sourceView is incorrect
```

Comment:
Tricky because did tap is open function. Considering deprecating this over a new version of didTap which also gives access to the source view.

Reply:
best to check with @nuno-vieira about this

Reply:
Do you see any issue with the cellForRow? 🤔 If this works fine, I would not change it for now

Reply:
<img width="996" alt="Screenshot 2024-08-26 at 11 43 13" src="https://github.com/user-attachments/assets/a9138cd7-1310-4c3c-9226-44f75aef0a12">
Kind of acceptable.

Reply:
Yeah I think it is not bad 🤔